ebos/ecloutputblackoilmodule.hh

```diff
@@ -234,11 +234,11 @@ public:
                 if (FluidSystem::phaseIsActive(oilPhaseIdx)) {
                     this->oilPressure_[globalDofIdx] = getValue(fs.pressure(oilPhaseIdx));
                 }else{
-                    // put pressure in oil pressure for output
-                    if (FluidSystem::phaseIsActive(waterPhaseIdx)) {
-                        this->oilPressure_[globalDofIdx] = getValue(fs.pressure(waterPhaseIdx));
-                    } else {
+                    // put pressure in oil pressure for output. Use gas if oil is not present
+                    if (FluidSystem::phaseIsActive(gasPhaseIdx)) {
                         this->oilPressure_[globalDofIdx] = getValue(fs.pressure(gasPhaseIdx));
+                    } else { // use water if neither oil nor gas is present
+                        this->oilPressure_[globalDofIdx] = getValue(fs.pressure(waterPhaseIdx));
                     }
```

Comment:
If we're changing this anyway, I think I'd prefer that we straighten out the conditionals too, e.g.,
```C++
if (FluidSystem::phaseIsActive(oilPhaseIdx)) {
    // ...
}
else if (FluidSystem::phaseIsActive(gasPhaseIdx)) {
    // ...
}
else {
    // ...
}
```
We might even consider writing this as a loop
```C++
for (const auto idx : { oilPhaseIdx, gasPhaseIdx, waterPhaseIdx }) {
    if (FluidSystem::phaseIsActive(idx)) {
        this->oilPressure_[globalDofIdx] = getValue(fs.pressure(idx));
        break;
    }
}
```
but that might on the other hand be too obscure.

Reply:
I think we should also rename oilPressure_. What about referencePressure_?

Reply:
> I think we should also rename `oilPressure_`

I think so too.  Explicitly referring to this as oil pressure is likely to be more confusing as time passes.

> What about `referencePressure_`?

Yeah, either that or maybe `fluidPressure_`.

Reply:
Good suggestion.